File: walletWeb3.js
if (typeof window.ethereum !== 'undefined') {
    const web3 = new Web3(window.ethereum);
    let userAddress;

    async function connectWallet() {
        const accounts = await window.ethereum.request({ method: 'eth_requestAccounts' });
        userAddress = accounts[0];
        console.log('Connected wallet address:', userAddress);
    }

    async function getBalance(address) {
        const tokenAddress = '0xYourTokenAddress'; // آدرس قرارداد توکن
        const abi = [
            {
                "constant": true,
                "inputs": [{"name": "_owner", "type": "address"}],
                "name": "balanceOf",
                "outputs": [{"name": "balance", "type": "uint256"}],
                "payable": false,
                "stateMutability": "view",
                "type": "function"
            }
        ];
        const contract = new web3.eth.Contract(abi, tokenAddress);
        const balance = await contract.methods.balanceOf(address).call();
        return web3.utils.fromWei(balance);
    }

    async function sendTokens(to, amount) {
        const tokenAddress = '0xYourTokenAddress'; // آدرس قرارداد توکن
        const abi = [
            {
                "constant": false,
                "inputs": [{"name": "to", "type": "address"}, {"name": "amount", "type": "uint256"}],
                "name": "transfer",
                "outputs": [{"name": "", "type": "bool"}],
                "payable": false,
                "stateMutability": "nonpayable",
                "type": "function"
            }
        ];
        const contract = new web3.eth.Contract(abi, tokenAddress);

        await contract.methods.transfer(to, web3.utils.toWei(amount, 'ether')).send({ from: userAddress });
    }

    connectWallet();
} else {
    alert('Please install MetaMask!');
}

File: walletWeb3.py
from web3 import Web3

# اتصال به شبکه Rinkeby یا هر شبکه دیگری
w3 = Web3(Web3.HTTPProvider('https://rinkeby.infura.io/v3/YOUR_INFURA_PROJECT_ID'))

# آدرس قرارداد توکن
token_address = '0xYourTokenAddress'
abi = [
    {
        "constant": True,
        "inputs": [{"name": "_owner", "type": "address"}],
        "name": "balanceOf",
        "outputs": [{"name": "balance", "type": "uint256"}],
        "payable": False,
        "stateMutability": "view",
        "type": "function"
    }
]

# اتصال به قرارداد
contract = w3.eth.contract(address=token_address, abi=abi)

# تابع برای دریافت موجودی
def get_balance(address):
    return contract.functions.balanceOf(address).call()

# آدرس کیف پول
address = '0xYourWalletAddress'
balance = get_balance(address)
print(f'Balance: {balance / 10**18} Tokens')  # موجودی به واحد اصلی
